test/zipkin/util.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//zipkin_util has constants and helper methods useful for zipkin tracing support.
+
+package zipkin
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/knative/pkg/test/logging"
+	"go.opencensus.io/trace"
+	"k8s.io/client-go/kubernetes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+
+	// ZipkinPortForwardPid environment variable name to store Process ID of the process started for zipkin port-foward.
+	ZipkinPortForwardPid = "ZIPKIN_PORTFWD_PID"
+
+	//ZipkinTraceIDHeader HTTP response header key to be used to store Zipkin Trace ID.
+	ZipkinTraceIDHeader = "ZIPKIN_TRACE_ID"
+
+	// ZipkinPort port exposed by the Zipkin Pod
+	ZipkinPort = 9411
+
+	// ZipkinTraceEndpoint port-forwarded zipkin endpoint
+	ZipkinTraceEndpoint = "http://localhost:9411/api/v2/trace/%s"
+
+	// ZipkinNamespace namespace where zipkin pod runs.
+	ZipkinNamespace = "istio-system"
```

Comment:
Why on `istio-system`? Can it be namespace agnostic? Can it run on a non-Knative namespace?

Reply:
From the comment here this is a limitation on the Istio side: https://github.com/knative/serving/blob/master/config/monitoring/200-common/100-zipkin.yaml#L19

Reply:
Thanks. Please clarify it here as well.

Reply:
Fixed.